File: 백준/Gold/14565. 역원（Inverse） 구하기/역원（Inverse） 구하기.py
def adding_reverse(a:int, modulo:int):
    return (-1 * a) % modulo

def multiplying_reverse(a:int, modulo:int):
    if gcd(a, modulo) != 1:
        return -1
    else:
        return eea(a, modulo, 1) % modulo

def gcd(x:int, y:int) -> int:
    x, y = max(x,y), min(x,y)
    if x%y:
        return gcd(y, x%y)
    else:
        return y

def eea(a:int, b:int, d:int) -> int:
    x_y_seq_memo:list[list[int]] = [[1,0],[0,1]]
    r_memo:list[int] = [a,b]
    q_i:int = 0
    i:int = 2
    while True:
        q_i = r_memo[(i-2)%2] // r_memo[(i-1)%2]
        r_memo[i%2]=r_memo[(i-2)%2] % r_memo[(i-1)%2]
        for j in range(2):
            x_y_seq_memo[i%2][j]=x_y_seq_memo[(i-2)%2][j]-x_y_seq_memo[(i-1)%2][j]*q_i
        if r_memo[i%2]==d:
            break
        else:
            i += 1
    return x_y_seq_memo[i%2][0]

N, A = list(map(int, input().split(" ")))
print(f'{adding_reverse(A, N)} {multiplying_reverse(A, N)}')

File: 백준/Gold/14565. 역원（Inverse） 구하기/역원（Inverse） 구하기.cc
#include <iostream>
#include <stdint.h>
#include <complex>
#include <cmath>
#include <vector>
#include <queue>
#include <set>
#include <map>
#include <algorithm>
#include <fstream>
#include <random>
#include <chrono>
#include <quadmath.h>
#include <type_traits>
#include <bit>
#include <bitset>
#include <numbers>

#pragma GCC optimize("O3")
#pragma GCC optimize("Ofast")
#pragma GCC optimize("unroll-loops")
#pragma GCC target("avx,avx2")

using namespace std;

using int128 = __int128_t;
using int64 = int64_t;
using int32 = int32_t;
using int16 = int16_t;
using int8 = int8_t;
using uint128 = __uint128_t;
using uint64 = uint64_t;
using uint32 = uint32_t;
using uint16 = uint16_t;
using uint8 = uint8_t;
using float80 = long double;
using float64 = double;
using float32 = float;
using complex80 = complex<float80>;
using complex64 = complex<float64>;
using complex32 = complex<float32>;
using str = string;
using None = void;
template <typename T>
using vec = std::vector<T>;
template <typename T, typename U>
using dict = std::map<T, U>;
template <typename T>
using set = std::set<T>;
template <typename T>
using heap = std::priority_queue<T>;
#define print cout
#define input cin
#define append push_back
#define appendleft push_front
#define pop pop_back
#define popleft pop_front

vector<int64_t> int64_extended_euclidean_algorithm(int64_t a, int64_t b, int64_t d)
{
    vector<vector<int64_t>> xy_sequence = {{1, 0}, {0, 1}};
    vector<int64_t> remainder_sequence = {a, b};
    int64_t quotient = 0;
    int64_t sequence_index = 2;
    while (true)
    {
        quotient = remainder_sequence[(sequence_index - 2) % 2] / remainder_sequence[(sequence_index - 1) % 2];
        remainder_sequence[sequence_index % 2] = remainder_sequence[(sequence_index - 2) % 2] - quotient * remainder_sequence[(sequence_index - 1) % 2];
        for (int64_t j = 0; j < 2; ++j)
        {
            xy_sequence[sequence_index % 2][j] = xy_sequence[(sequence_index - 2) % 2][j] - xy_sequence[(sequence_index - 1) % 2][j] * quotient;
        }
        if (remainder_sequence[sequence_index % 2] == d)
        {
            break;
        }
        ++sequence_index;
    }
    return xy_sequence[sequence_index % 2];
}

int64_t adding_reverse(int64_t a, int64_t mod)
{
    return (-a % mod + mod) % mod;
}

int64_t multiplying_reverse(int64_t a, int64_t mod)
{
    if (gcd(a, mod) != 1)
    {
        return -1;
    }
    return ((int64_extended_euclidean_algorithm(a, mod, 1)[0] % mod) + mod) % mod;
}

None fastIO()
{
    std::cin.tie(NULL);
    std::cout.tie(NULL);
    std::ios_base::sync_with_stdio(false);
}

int32 main()
{
    fastIO();
    int64 n, a;
    input >> n >> a;
    print << adding_reverse(a, n) << " " << multiplying_reverse(a, n);
}